### PR TITLE
Concision contribution.

### PR DESCRIPTION
**Pull request by paulp**
Updates to use more direct/concise types.

### CODE REVIEW

File: src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala [hunk 1]
```diff
@@ -505,7 +500,7 @@ abstract class SpecializeTypes extends InfoTransform with TypingTransformers {
    *  the original class, leaves everything else as-is.
    */
   private def mapAnyRefsInOrigCls(env: TypeEnv, origcls: Symbol): TypeEnv = env map {
-    case (sym, tp) if (tp == AnyRefClass.tpe) && sym.owner == origcls => (sym, sym.tpe)
+    case (sym, tp) if (tp == AnyRefTpe) && sym.owner == origcls => (sym, sym.tpe)
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala]:
``` scala
case (sym, AnyRefTpe) if sym.owner == origcls => (sym, sym.tpe)
```

Comment (paulp) [inline src/compiler/scala/tools/nsc/transform/SpecializeTypes.scala]: Indeed, I didn't really seek out the "second order simplifications" which this must have enabled in pattern matches.

**Comment by paulp** [comment]
All I changed in the re-push was moving the test to the second commit; it passes for me, so it must depend on the second commit as well as the first.

**Comment by paulp** [comment]
By the power of Grayskull, I merge thee!

**Pull request merged**